notary-integration-test/src/integration-test/kotlin/integration/eth/EthTokensProviderTest.kt

```diff
@@ -29,18 +33,51 @@ class EthTokensProviderTest {
         val expectedTokens = mutableMapOf<String, EthTokenInfo>()
         (1..tokensToAdd).forEach { i ->
             val ethWallet = "0x$i"
-            val tokenInfo = EthTokenInfo(String.getRandomString(9), i.toShort())
-            expectedTokens[ethWallet] = tokenInfo
-            integrationHelper.addERC20Token(ethWallet, tokenInfo.name, tokenInfo.precision)
+            val tokenName = String.getRandomString(9)
+            val tokenPrecision = i.toShort()
+            expectedTokens[ethWallet] = EthTokenInfo(tokenName, tokenPrecision)
+            integrationHelper.addERC20Token(ethWallet, tokenName, tokenPrecision)
         }
         ethTokensProvider.getTokens()
             .fold(
                 { tokens ->
                     assertFalse(tokens.isEmpty())
                     expectedTokens.forEach { (expectedEthWallet, expectedTokenInfo) ->
-                        assertEquals(expectedTokenInfo, tokens.get(expectedEthWallet))
+                        val (expectedName, expectedPrecision) = expectedTokenInfo
+                        assertEquals(expectedName, tokens.get(expectedEthWallet))
+                        assertEquals(expectedPrecision, ethTokensProvider.getTokenPrecision(expectedName).get())
+                        assertEquals(expectedEthWallet, ethTokensProvider.getTokenAddress(expectedName).get())
                     }
                 },
                 { ex -> fail("Cannot get tokens", ex) })
     }
+
+    /**
+     * @given Iroha network is running
+     * @when tokenProvider is queried with some nonexistent asset
+     * @then failure result is returned
+     */
+    @Test
+    fun getNonexistentTokenn() {
+        ethTokensProvider.getTokenPrecision("nonexist")
+            .success { fail("Result returned success while failure is expected.") }
+
+        ethTokensProvider.getTokenAddress("nonexist")
+            .success { fail("Result returned success while failure is expected.") }
+    }
+
+    /**
+     * Test predefined asset ethereum.
+     * @given Iroha network is running
+     * @when tokenProvider is queried with "ether"
+     * @then predefined parameters are returned
+     */
+    @Test
+    fun getEthereum() {
+        ethTokensProvider.getTokenPrecision(ETH_NAME)
+            .success { assertEquals(ETH_PRECISION, it) }
```

Comment:
I think we have to call fail() in case of failure. Otherwise junit will consider this test passed.

Reply:
done